### DISCUSSION TITLE
Allow to customize permissions of API Token

### DISCUSSION THREAD

**Discussion by tsimbalar** [codemagic-ci-cd/codemagic-docs#1731] [Ideas]
### Use case

```Text
We are using our personal API Token to generate a dashboard that shows the status of the builds of our app in CodeMagic. 

Unfortunately, we have to use the only Token that we have, which has "all the privileges" the current user has.

We would like to be able to have a token with only Read Access, in order to limit the possible risks (a full token enables to trigger a new build, implicitly allowing to publish apps to the App Store etc)
```

### Idea

```Text
Allow to customize the "scope" of the API tokens we get from the Web UI.
```

**Comment by Ycoi** [comment]
@tsimbalar Thanks for the idea and for a very clear description of your use case and requirements! 

We’ll discuss this in the Codemagic team and see what we can do and what the technical implications would be. I’ll get back to you when I know more.

**Reply by Ycoi** [reply]
@tsimbalar Sorry that I only now come back to this. We discussed this with the team and we unfortunately do not have immediate plans on adding this functionality. 

I do think it’s a very good idea though so I have marked it down and we will be getting back to this as a part of further improvements to the authentication and the Codemagic API in general. We unfortunately don't have an estimate when that would happen though, but the best way to keep up to date on new features and product changes is to keep an eye on the announcement section of our GitHub Discussions.

**Comment by Ycoi** [comment]
I will close this thread since we unfortunately don’t have plans to add this feature in the foreseeable future. Thanks for the idea though and if you have any other ideas or future requests, please don’t hesitate to open a new thread.
